### DISCUSSION TITLE
Meaning of using "or" in license text

### DISCUSSION THREAD

**Discussion by lanrebr** [jakartaee/persistence#446] [Q&A] [answered]
Could the community explain the meaning of specifying the two licenses used to release the code?

For example, in https://github.com/jakartaee/persistence/commit/cb98bfa81fbd8b493bc3532faa9e450f8af239b1 we see that the copyright is been updated to indicate that the code is licensed under EPL 2.0 or EDL 1.0:

Copyright (c) YYYY Oracle and/or its affiliates.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0,
 * **or** the Eclipse Distribution License v. 1.0 which is available at
 * http://www.eclipse.org/org/documents/edl-v10.php.
 *
 * SPDX-License-Identifier: EPL-2.0 **OR** BSD-3-Clause

The concern is that in EPL 2.0 license the commercial use text reads that:
"...hereby agrees to defend and indemnify every other Contributor (“Indemnified Contributor”) against any losses, damages and costs (collectively “Losses”) arising from claims, lawsuits and other legal actions brought by a third party against the Indemnified Contributor to the extent caused by the acts or omissions of such Commercial Contributor in connection with its distribution of the Program in a commercial product offering."

whereas EDL 2.0 imposes no such requirements or conditions, seems to allow real free and open use of the code.

Does the "or" means that both licenses are imposed or one of them is imposed? I would imagine the least restrictive one applies. Why there is need to have both? 

If this clause precludes people from using and distributing the software for commercial purposes, should the EPL 2.0 not be removed?

**Comment by lukasj** [comment]
**OR** in this context is a disjunctive operator meaning that either EPL-2.0 or BSD-3-Clause applies, but not both at the same time, see [the SPDX spec](https://spdx.github.io/spdx-spec/v2-draft/SPDX-license-expressions/)

But I am not a lawyer, so use my answer with caution. Should you need really funded answer, I recommend directing this question to _license (at) eclipse.org_

**Comment by lanrebr** [comment]
Well that is the question, given that the licenses are in contradiction
which each other. Say for example that EPL requires the user to release the
source code and BSD does not require the user to do it, means that the user
can choose not to release the code?

And given that BSD is more lax than EPL why even include EPL at all?

**Comment by lukasj** [comment, marked as answer]
choose the one which suits you the best and comply with its rules. If you prefer EDL, choose that and comply with its rules. If you prefer EPLv2, choose EPLv2 and comply with its rules.

Based on https://www.eclipse.org/legal/epl-2.0/faq.php my answer to _why having dual license_ would be that the reason lies in having some level of compatibility with GPL license. But as I said - these questions are better suited for lawyers at _license (at) eclipse.org_

**Reply by lanrebr** [reply]
Thank you for the https://www.eclipse.org/legal/epl-2.0/faq.php#h.a0eux401qus link. It seems that is not really a choice, but it requires and depends on what license is used in other projects used along with it. If there are no other projects used then the EPL license stands otherwise the license of the other project stands.
